### PR TITLE
Overwatch refactor

### PR DESCRIPTION
**Pull request by ThePiachu**
# About the pull request

While working on #1910 I noticed three places had very similar copy-pasted squad info code, so I decided to refactor them into one place. It contains a fair number of differnet strings for it to work and be formatted just the way a given UI wants it, but at least the logic stays consistent...

# Explain why it's good for the game

Less redundant code more better.

# Testing Photographs and Procedure

<details>
<summary>Screenshots & Videos</summary>

Put screenshots and videos here with an empty line between the screenshots and the `<details>` tags.

</details>

# Changelog

:cl:
code: Refactored the code around listing squad members for the Overwatch Console, Crew Monitor and the Groundside Operations Computer.
/:cl:

### CODE REVIEW

**Comment by fira** [comment]
Finally!
This sounds a bit redundant with TGUI'ifing the Overwatch consoles though :(

**Comment by ThePiachu** [comment]
Well, it's not only Overwatch but two other places that use it. And yeah, once we have a proper TGUI chances are these functions will be converted to support the other data needed.

File: code/__DEFINES/machinery.dm [hunk 1]
```diff
@@ -5,3 +5,9 @@
 #define USE_POWER_NONE		0
 #define USE_POWER_IDLE		1
 #define USE_POWER_ACTIVE 	2
+
+// Overwatch defines
+
+#define OVERWATCH_HIDE_ALMAYER 2
+#define OVERWATCH_HIDE_GROUND 1
+#define OVERWATCH_HIDE_NONE 0
```
Comment (harryob) [inline code/__DEFINES/machinery.dm:13]: no eol at the end of your file - are you using the editorconfig plugin btw?

File: code/game/jobs/job/marine/squads.dm [hunk 2]
```diff
@@ -765,3 +765,248 @@
...
+	var/smart_text_groundside_operations = ""
+	var/marine_text_groundside_operations = ""
+	var/misc_text_groundside_operations = ""
+
+	var/SL_z //z level of the Squad Leader
+	if(squad_leader)
+		var/turf/SL_turf = get_turf(squad_leader)
+		SL_z = SL_turf.z
+
+	for(var/X in marines_list)
```
Comment (harryob) [inline code/game/jobs/job/marine/squads.dm:846]: needs a more descriptive var name... possible_squad_member... or something

File: code/game/jobs/job/marine/squads.dm [hunk 3]
```diff
@@ -765,3 +765,248 @@
...
+		var/dist = "<b>???</b>"
+		var/area_name = "<b>???</b>"
+		var/mob/living/carbon/human/H
+
+		var/is_filtered = FALSE
+		if(X && ("\ref[X]" in marine_filter))
+			is_filtered = TRUE
+
+		if(ishuman(X))
+			H = X
```
Comment (harryob) [inline code/game/jobs/job/marine/squads.dm:863]: pain (ik none of this is yours)

File: code/game/jobs/job/marine/squads.dm [hunk 4]
```diff
@@ -765,3 +765,248 @@
...
+		if(!X)
+			continue //just to be safe
+		var/mob_name = "unknown"
+		var/mob_state = ""
+		var/role = "unknown"
+		var/act_sl = ""
+		var/fteam = ""
+		var/dist = "<b>???</b>"
+		var/area_name = "<b>???</b>"
+		var/mob/living/carbon/human/H
```
Comment (harryob) [inline code/game/jobs/job/marine/squads.dm:856]: human and should also be after the typecheck

**Review by harryob — CHANGES_REQUESTED** [review]

File: code/game/jobs/job/marine/squads.dm (hunk 4 above)
Comment (ThePiachu) [reply code/game/jobs/job/marine/squads.dm]: Some code still references the /mob outside the if statement. Probably should be fixed, but I didn't want to change the logic, only to squish the code.

File: code/game/jobs/job/marine/squads.dm [hunk 5]
```diff
@@ -765,3 +765,248 @@
...
+
+		if(ishuman(marine))
+			human_marine = marine
+			mob_name = human_marine.real_name
+			var/area/A = get_area(human_marine)
+			var/turf/M_turf = get_turf(human_marine)
+			if(!M_turf)
+				continue
+			if(A)
+				area_name = sanitize_area(A.name)
```
Comment (harryob) [inline code/game/jobs/job/marine/squads.dm:870]:
```suggestion
			var/area/marine_area = get_area(human_marine)
			var/turf/marine_turf = get_turf(human_marine)
			if(!marine_turf)
				continue
			if(marine_area)
				area_name = sanitize_area(marine_area.name)
```

File: code/game/jobs/job/marine/squads.dm [hunk 6]
```diff
@@ -765,3 +765,248 @@
...
+			if(human_marine.job)
+				role = human_marine.job
+			else if(istype(human_marine.wear_id, /obj/item/card/id)) //decapitated marine is mindless,
+				var/obj/item/card/id/ID = human_marine.wear_id		//we use their ID to get their role.
+				if(ID.rank) role = ID.rank
+
+			if(user)
+				var/turf/user_turf = get_turf(user)
+				if(M_turf)
+					var/area/mob_area = M_turf.loc
```
Comment (harryob) [inline code/game/jobs/job/marine/squads.dm:889]:
```suggestion
				if(marine_turf)
					var/area/mob_area = marine_turf.loc
```
should probably just use a get_area here tho

**Review by harryob — CHANGES_REQUESTED** [review]

**Review by harryob — APPROVED** [review]

**Pull request merged**